NitroxClient/MonoBehaviours/NitroxEntity.cs

```diff
@@ -14,6 +15,8 @@ public class NitroxEntity : MonoBehaviour, IProtoTreeEventListener
     {
         private static Dictionary<NitroxId, GameObject> gameObjectsById = new Dictionary<NitroxId, GameObject>();
 
+        private static Dictionary<NitroxId, List<GameObject>> gameObjectsWaitingForParent = new Dictionary<NitroxId, List<GameObject>>();
```

Comment:
Let's move this logic out into a new class.  We can call it right before spawning a new entity.